provider/metrics/container_insights.go

```diff
@@ -0,0 +1,315 @@
+	wrapper := &containerInsightsMetricsWrapper{
+		cpuMetricsSeries:    make(map[string][]aci.TimeSeriesEntry),
+		memoryMetricsSeries: make(map[string][]aci.TimeSeriesEntry),
+	}
+	systemStats, err := metricsGetter.GetContainerGroupMetrics(ctx, resourceGroup, containerGroup, aci.MetricsRequest{
+		Dimension:    "containerName eq '*'",
+		Start:        start,
+		End:          end,
+		Aggregations: []aci.AggregationType{aci.AggregationTypeAverage},
+		Types:        []aci.MetricType{aci.MetricTypeCPUUsage, aci.MetricTypeMemoryUsage},
+	})
+	if err != nil {
+		return nil, errors.Wrapf(err, "error fetching cpu/mem stats for container group %s", containerGroup)
+	}
+
+	netStats, err := metricsGetter.GetContainerGroupMetrics(ctx, resourceGroup, containerGroup, aci.MetricsRequest{
+		Start:        start,
+		End:          end,
+		Aggregations: []aci.AggregationType{aci.AggregationTypeAverage},
+		Types:        []aci.MetricType{aci.MetricTyperNetworkBytesRecievedPerSecond, aci.MetricTyperNetworkBytesTransmittedPerSecond},
+	})
+	if err != nil {
+		return nil, errors.Wrapf(err, "error fetching network stats for container group %s", containerGroup)
+	}
+
+	logger.Debug("container_insights.queryContainerInsightsMetrics, systemStats: %v", systemStats)
+	logger.Debug("container_insights.queryContainerInsightsMetrics, netStats: %v", netStats)
+
+	for _, metrics := range systemStats.Value {
+		if metrics.Desc.Value == aci.MetricTypeCPUUsage {
+			for _, timeSeries := range metrics.Timeseries {
+				containerNameMeta := funk.Find(timeSeries.MetadataValues, func(meta aci.MetricMetadataValue) bool {
+					return meta.Name.Value == "containername"
+				})
+				if containerNameMeta != nil {
+					containerName := containerNameMeta.(aci.MetricMetadataValue).Value
+					wrapper.cpuMetricsSeries[containerName] = timeSeries.Data
+				}
+			}
+		}
+		if metrics.Desc.Value == aci.MetricTypeMemoryUsage {
+			for _, timeSeries := range metrics.Timeseries {
+				containerNameMeta := funk.Find(timeSeries.MetadataValues, func(meta aci.MetricMetadataValue) bool {
+					return meta.Name.Value == "containername"
+				})
+				if containerNameMeta != nil {
+					containerName := containerNameMeta.(aci.MetricMetadataValue).Value
+					wrapper.memoryMetricsSeries[containerName] = timeSeries.Data
+				}
+			}
+		}
+	}
+
+	for _, metrics := range netStats.Value {
+		if metrics.Desc.Value == aci.MetricTyperNetworkBytesRecievedPerSecond && len(metrics.Timeseries) > 0 {
+			wrapper.networkRxMetricSeries = metrics.Timeseries[0].Data
+		}
+		if metrics.Desc.Value == aci.MetricTyperNetworkBytesTransmittedPerSecond && len(metrics.Timeseries) > 0 {
+			wrapper.networkTxMetricSeries = metrics.Timeseries[0].Data
+		}
+	}
+	return wrapper, nil
+}
+
+func (wrapper *containerInsightsMetricsWrapper) getCPUMetricsSeries(containerName string) ([]aci.TimeSeriesEntry, bool) {
```

Comment:
`getGPU`

Reply:
There are many information in Container Insight's result: metrics's meta data, time series's meta data and multiple time series data.  Here is to get the time series data for CPU. So the long name "getCPUMetricsSeries" more meaningful and easier for understanding the codes.

Reply:
fixed that.